lightningd/plugin.c

```diff
@@ -770,6 +774,47 @@ static bool plugin_rpcmethods_add(struct plugin *plugin,
 	return true;
 }
 
+static bool plugin_subscriptions_add(struct plugin *plugin, const char *buffer,
+				     const jsmntok_t *resulttok)
+{
+	const jsmntok_t *subscriptions =
+	    json_get_member(buffer, resulttok, "subscriptions");
+
+	if (!subscriptions) {
+		plugin->subscriptions = NULL;
+		return true;
+	}
+	plugin->subscriptions = tal_arr(plugin, char *, 0);
+	if (subscriptions->type != JSMN_ARRAY) {
+		plugin_kill(plugin, "\"result.subscriptions\" is not an array");
+		return false;
+	}
+
+	for (int i = 0; i < subscriptions->size; i++) {
+		char *topic;
+		const jsmntok_t *s = json_get_arr(subscriptions, i);
+		if (s->type != JSMN_STRING) {
+			plugin_kill(
+			    plugin,
+			    "result.subscriptions[%d] is not a string: %s", i,
+			    plugin->buffer);
+			return false;
+		}
+		topic = tal_strndup(
+		    plugin, plugin->buffer + s->start, s->end - s->start);
+
+		if (!notifications_have_topic(topic)) {
+			plugin_kill(
+			    plugin,
+			    "topic %s is not a know notification topic", topic);
```

Comment:
s/know/known/

Reply:
Slight simplification, I would skip the 's->type != JSMN_STRING' check above, and rely on it here (in which case, please put  ' around the %s).